src/charm.py

```diff
@@ -234,14 +242,14 @@ def _check_config(self) -> Tuple[str, str]:
         except ChangeError as e:
             output, err = "", e.err
         except ExecError as e:
-            output, err = e.stdout, e.stderr
+            output, err = str(e.stdout), str(e.stderr)
```

Comment:
Do we need `.decode().strip()` here? If it's coming directly from subprocess then stdout, stderr are bytes.

Reply:
That's why I'd assume this is fine :) If you agree please mark the conversation as resolved!